registry.go

```diff
@@ -1,13 +1,6 @@
 package distribution
 
-import (
-	"io"
-	"time"
-
-	"github.com/docker/distribution/digest"
-	"github.com/docker/distribution/manifest"
-	"golang.org/x/net/context"
-)
+import "golang.org/x/net/context"
```

Comment:
@jlhawn The idea here is that we subscribe to `x/net/context` or we don't. Really, each method should start with a context argument. Perhaps, we could refactor this interface to be transaction oriented to capture that, making context an implementation detail of the transaction.